src/store/user/slice.js

```diff
@@ -0,0 +1,27 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const userSlice = createSlice({
+  name: "user",
+  initialState: {
+    status: "no data",
+    data: null,
+    error: null,
+  },
+  reducers: {
+    setUserError(state, action) {
+      state.status = "error";
+      state.data = null;
+      console.log(action);
+    },
+    setUserLoaded(state, action) {
```

Comment:
it's better to name it `setUser`, because everything this reducer should do is just putting the user to redux store and you can have another reducer for `state.status`

it makes reducers more reusable

statuses should be taken from constants